designdocs/passkey_p256.md

```diff
@@ -0,0 +1,555 @@
+    - [Separate Pallet](#separate-pallet)
+  - [8. References](#8-references)
+
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on the Frequency chain. Passkey support aims to provide a novel non-custodial solution for managing user accounts and signing transactions on-chain.
+
+## 2. Benefits
+
+- **Non-Custodial Account Management**: Passkey support enables users to interact with the Frequency chain without actual seed access. This reduces the risk of seed exposure and enhances security.
+
+- **Functionality**: Passkey support provides a better user experience by enabling users to sign transactions without seed access. The proliferation of Passkey-enabled services can lead to a seamless user experience.
+
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account Public Key, users can prove ownership of the account and recover access in case of a lost seed.
+
+## 3. Terminology
+
+### Accounts
+
+- **Passkey Account**: The user account associated with the Passkey.
+- **Frequency Account**: The account used for on-chain transactions and interactions. This account is derived from a seed phrase generated by the Web Wallet.
+- **Web Wallet**: The platform facilitating the management and interaction with user accounts.
+
+### Keys
+
+- **Passkey**: P256 key pair used for transaction signing and account management. This is the primary key used for transaction signing.
+- **Passkey Public Key**: The Public Key retrieved from the Passkey registration process.
+- **Account Key**: The key pair generated from a seed phrase.
+- **Account Public Key**: The Public Key derived from the Account Key.
+
+### Signatures
+
+- **Account signature on Passkey Public Key**: A cryptographic signature generated using the Account Key. The data being signed is the Passkey Public Key. This is used to prove ownership of Account Key.
+- **Passkey Signatures**: A cryptographic signature generated using the Passkey private key. This is presented to Passkey enabled services as a challenge-response mechanism. Passkeys are used to generate two signatures as follows:
+  - **Signature on Account Public Key**: Passkey signs a message containing the account Public Key. This signature is retained by a Web Wallet platform and/or maybe used for account recovery.
+  - **Signature on Transactions**: Passkey signs the transaction payload which needs to be submitted on-chain. This signature is used to verify the authenticity of the transaction.
+
+## 4.1. Data Flow Diagram
+
+- Simplified Registration Diagram
```

Comment:
These diagram links are broken for me, even when signed into my Google account.

Reply:
Odd. They worked for me when they switched, but not anymore.

@aramikm if they were in drawio this might help: https://www.drawio.com/blog/embed-diagrams-github-markdown

Reply:
I'll take a look